micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java

```diff
@@ -79,6 +79,11 @@ public void connectionPoolClosed(ConnectionPoolClosedEvent event) {
         waitQueueSize.remove(serverId);
     }
 
+//    @Override
```

Comment:
What metrics would you suggest we make related to these events?

Reply:
I'm not completely sure - i have not thought too much about it. I would reword my comment slightly...

> I am in favor of implementing these.

to "I am in favor of investigating these events a bit and seeing if it makes sense to add them".

It could also be done on a follow on PR once the base 3.x->4.x change is in.

Reply:
Yepp, I saw them I'm just not sure if we should register any meter for them.
Maybe counting the `connectionCheckOutFailed` could be useful or a new counter on the `connectionCheckedOut`-`connectionCheckOutFailed` axis with different tags.

Reply:
I am in favor of splitting the "new" metrics off into a separate ticket. What do you think @jonatan-ivanov ?

Reply:
👍🏼 Feel free to create an issue. If you also want to PR it just let me know, I can assign it to you.

Reply:
@jonatan-ivanov #2499

Thank you for the offer to assign it to me but I am not sure when I would get to it as I have my hands in a few cookie jars at the same time currently. :)

Reply:
@jonatan-ivanov #2499

Thank you for the offer to assign it to me but I am not sure when I would get to it as I have my hands in a few cookie jars at the same time currently. :)